lua/telescope/pickers/entry_display.lua

```diff
@@ -1,15 +1,27 @@
+local utils = require('telescope.utils')
+
 local entry_display = {}
 
 local function truncate(str, len)
   str = tostring(str) -- We need to make sure its an actually a string and not a number
-  -- TODO: This doesn't handle multi byte chars...
-  if vim.fn.strdisplaywidth(str) > len then
-    str = str:sub(1, len - 1)
-    str = str .. "…"
+  if utils.strdisplaywidth(str) <= len then
+    return str
+  end
+  local charlen = 0
+  local result = ''
+  while true do
+    local part = utils.strcharpart(str, 0, charlen) ..  '…'
+    if utils.strdisplaywidth(part) > len then
+      break
+    end
+    result = part
+    charlen = charlen + 1
   end
-  return str
+  return result
 end
 
+entry_display.truncate = truncate
```

Comment:
I like to have `entry_display.create` immediately rather then first creating a `local function`

Reply:
Sure. fixed on 19188ce